src/Automation/Scanner.cs

```diff
@@ -23,58 +27,22 @@ internal Scanner(Config config, IScanTools scanTools)
             if (scanTools.OutputFileHelper == null) throw new ArgumentException(ErrorMessages.ScanToolsOutputFileHelperNull, nameof(scanTools));
         }
 
-        /// <summary>
-        /// See <see cref="IScanner.Scan()"/>
-        /// </summary>
-        /// <returns></returns>
-        public ScanResults Scan()
+        public Task<ScanOutput> ScanAsync(ScanOptions scanOptions, CancellationToken cancellationToken)
         {
-            if (_config.AreMultipleScanRootsEnabled)
-            {
-                throw new InvalidOperationException("Multiple scan roots are not supported when calling Scan(). Use ScanAll() instead.");
-            }
-            return ExecuteScan(null).First();
+            HandleScanOptions(scanOptions);
+            return SnapshotCommand.ExecuteAsync(_config, _scanTools, cancellationToken);
         }
 
-        /// <summary>
-        /// See <see cref="IScanner.ScanAll()"/>
-        /// </summary>
-        /// <returns></returns>
-        public IReadOnlyCollection<ScanResults> ScanAll()
+        public ScanOutput Scan(ScanOptions scanOptions)
         {
-            return ExecuteScan(null);
+            HandleScanOptions(scanOptions);
+            return SnapshotCommand.Execute(_config, _scanTools);
         }
 
-        /// <summary>
-        /// See <see cref="IScanner.Scan(string)"/>
-        /// </summary>
-        /// <returns></returns>
-        public ScanResults Scan(string scanId)
+        private void HandleScanOptions(ScanOptions scanOptions)
         {
-            if (_config.AreMultipleScanRootsEnabled)
-            {
-                throw new InvalidOperationException("Multiple scan roots are not supported when calling Scan(). Use ScanAll() instead.");
-            }
-            return ExecuteScan(scanId).First();
-        }
-
-        /// <summary>
-        /// See <see cref="IScanner.ScanAll(string)"/>
-        /// </summary>
-        /// <returns></returns>
-        public IReadOnlyCollection<ScanResults> ScanAll(string scanId)
-        {
-            return ExecuteScan(scanId);
-        }
-
-        private IReadOnlyCollection<ScanResults> ExecuteScan(string scanId)
-        {
-            return ExecutionWrapper.ExecuteCommand(() =>
```

Comment:
Are we okay with no longer having the error handling that the execution wrapper provided?

Reply:
Yes